src/main/java/com/cloudbees/jenkins/support/api/TruncatedContent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2015 schristou88
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.support.api;
+
+import com.cloudbees.jenkins.support.timer.FileListCapComponent;
+
+import java.io.*;
+
+/**
+ * Content added to the support bundle that should be truncated.
+ *
+ * @author schristou88
+ */
+public abstract class TruncatedContent extends Content {
+  private int maxSize;
+
+  public TruncatedContent(String name) {
+    this(name, FileListCapComponent.MAX_FILE_SIZE);
+  }
+
+  public TruncatedContent(String name, int maxSize) {
+    super(name);
+    this.maxSize = maxSize;
+  }
+
+  @Override
+  public void writeTo(OutputStream os) throws IOException {
+    PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(new TruncatedOutputStream(os, maxSize), "UTF-8")));
+    try {
+      printTo(out);
+    } finally {
+      out.flush();
+    }
+  }
+
+  protected abstract void printTo(PrintWriter out) throws IOException;
```

Comment:
as this is intended to be implemented in sub-classes a little javadoc would go a long way...